server/models/User.js

```diff
@@ -292,10 +298,45 @@ User.getCounts = async function (teamId: string) {
   return {
     active: parseInt(counts.activeCount),
     admins: parseInt(counts.adminCount),
+    viewers: parseInt(counts.viewerCount),
     all: parseInt(counts.count),
     invited: parseInt(counts.invitedCount),
     suspended: parseInt(counts.suspendedCount),
   };
 };
 
+User.prototype.demote = async function (teamId: string, to: string) {
```

Comment:
I think we need tests for this method now, currently it's not going to work as expected as it assumes you're trying to demote an `admin` to a lower role, but now you could be demoting a member to a viewer – it would stop you doing that if there was only one admin on the team 😉

Reply:
Actually, it working! We are validating if there exists at least one admin in the team while not considering the rank of the current user.

Reply:
We can remove `at least one admin is required` validation if in the policies we can change to not let a user act on himself. 

This is confusing because in the app I cannot take any action on ourselves but I can through the API.

Reply:
Resolving the conversation because the function works as expected.